examples/dark-mode/app/utils/theme.server.ts

```diff
@@ -0,0 +1,31 @@
+import { createCookieSessionStorage } from "remix";
+
+import { Theme, isTheme } from "./theme-provider";
+
+// Make use to set the environment variable SESSION_SECRET before running the code
+const sessionSecret = process.env.SESSION_SECRET ?? 'DEFAULT_SECRET';
+
+const themeStorage = createCookieSessionStorage({
+  cookie: {
+    name: "my_remix_theme",
+    secure: true,
+    secrets: [sessionSecret],
+    sameSite: "lax",
+    path: "/",
+    httpOnly: true
+  }
+});
+
+async function getThemeSession(request: Request) {
+  const session = await themeStorage.getSession(request.headers.get("Cookie"));
+  return {
+    getTheme: () => {
+      const themeValue = session.get("theme");
+      return isTheme(themeValue) ? themeValue : Theme.LIGHT;
```

Comment:
This should return the theme in the session or `null`. If we just default to `Theme.LIGHT` then we'll never take the user's preference into account when they arrive on the site.

```suggestion
      return isTheme(themeValue) ? themeValue : null;
```

I tried doing this locally and had some funny stuff happen where it looked like some things showed up twice. I'm not sure what was causing that. Hopefully you don't see the same thing and it was just a weird thing on my machine or something.